extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/transaction/interceptor/MongoTransactionalInterceptor.java

```diff
@@ -0,0 +1,136 @@
+package io.quarkus.mongodb.panache.transaction.interceptor;
+
+import javax.annotation.Priority;
+import javax.inject.Inject;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.RollbackException;
+import javax.transaction.Status;
+import javax.transaction.Synchronization;
+import javax.transaction.SystemException;
+import javax.transaction.TransactionManager;
+import javax.transaction.TransactionSynchronizationRegistry;
+
+import com.mongodb.ReadConcern;
+import com.mongodb.ReadConcernLevel;
+import com.mongodb.ReadPreference;
+import com.mongodb.TransactionOptions;
+import com.mongodb.WriteConcern;
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.MongoClient;
+
+import io.quarkus.mongodb.panache.transaction.MongoTransactionException;
+import io.quarkus.mongodb.panache.transaction.Transactional;
+
+@Interceptor
+@Transactional
+@Priority(Interceptor.Priority.PLATFORM_BEFORE + 200)
+public class MongoTransactionalInterceptor {
+    private static final Object transactionKey = new Object();
+
+    @Inject
+    MongoClient client;
+
+    @Inject
+    TransactionSynchronizationRegistry tsr;
+
+    @Inject
+    TransactionManager transactionManager;
+
+    @AroundInvoke
+    public Object intercept(InvocationContext ic) throws Exception {
```

Comment:
At the very least you need to support the reactive return types with the same handling as in `TransactionalInterceptorBase.invokeInOurTx`, especially because you have a reactive client for mongo.

Reply:
Reactive MongoDB with Panache didn' exist at the time I wrote this PR.
I'll take a look

Reply:
Done